docusaurus/docs/Android/04-compose/04-message-components/03-message-list.mdx

```diff
@@ -3,50 +3,65 @@
 
 * **Bound**: This version binds itself to the `MessageListViewModel` and loads all the required data. It also connects single and long item tap, pagination, and bottom reached events to the `ViewModel`.
-* **Stateless**: This is a stateless version of the list, which doesn't know about the `ViewModel` and depends on pure state from external sources, to render its UI.
+* **Stateless**: This is a stateless version of the list which doesn't know about the `ViewModel` and depends on pure state from external sources in order to render its UI.
 
 :::note 
-The **bound** version of the list uses the **stateless** list internally. That way, when providing the same state to either component, the behavior will be the same. 
+The **bound** version of the list uses the **stateless** list internally. That way, when providing the same state to either component, the behavior remains the same.
 
 Additionally, we cannot provide a default `ViewModel` for this component, as it requires the `channelId` to load the data, so you'll have to build an instance yourself.
 :::
 
 Based on the provided state, this component shows the following UI:
 
-* `LoadingView`: If we're loading the initial data.
-* `EmptyView`: If there is no data and we've finished loading.
-* `Messages`: The list of messages in the channel, including file and image attachments, with various actions like thread clicks, item long taps, pagination and reaching the bottom.
+* **Loading indicator**: While we're loading the initial data.
+* **Empty content**: If there is no data and we've finished loading.
+* **Messages**: The list of messages in the channel, including file and image attachments, with various actions like thread clicks, item long taps, pagination and reaching the bottom.
 
 Let's see how to show a list of messages.
 
 ## Usage
 
-To use the **bound** `MessageList`, add it to the rest of your UI, e.g. within `setContent()`:
+To use the **bound** `MessageList`, first initialize the `ViewModel` using our `MessagesViewModelFactory`:
+
+```kotlin
+val factory by lazy {
+    MessagesViewModelFactory(
+        context = this,
+        channelId = channelId,
+    )
+}
+
+val listViewModel by viewModels<MessageListViewModel>(factoryProducer = { factory })
+```
+
+Then add it to the rest of your UI, e.g. within `setContent()`:
 
 ```kotlin
 override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     // Load data
 
     setContent {
-        Column(
-            Modifier.fillMaxSize(),
-            verticalArrangement = Arrangement.Top
-        ) {
-            MessageListHeader(...)
-
-            MessageList(
-                viewModel = listViewModel,
-                modifier = Modifier.fillMaxSize()
-            )
-
-            // Rest of your UI
+        ChatTheme {
+            Column(
+                Modifier.fillMaxSize(),
+                verticalArrangement = Arrangement.Top
```

Comment:
Why do we need `Arrangement.Top` here.

Reply:
Good question, I just checked and that's the default argument in `Column` so it seems it's quite unnecessary.

I'll make sure that we're able to remove it without side effects (check if we did something differently down the line) and then do so.